server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/routes/MailRepositoriesRoutesTest.java

```diff
@@ -466,71 +460,94 @@ public void retrievingAMailShouldDisplayItsInformation() throws Exception {
+
+        MailAddress recipientHeaderAddress = new MailAddress("third@party");
+        FakeMail mail = FakeMail.builder()
             .name(name)
-            .attributes(attributes)
+            .attribute("name1", "value1")
+            .attribute("name2", "value2")
             .mimeMessage(mimeMessage)
-            .size(messageSize);
-
-        for (String recipient : perRecipientHeaders.keySet()) {
-            MailAddress recipientKey = new MailAddress(recipient);
-            Map<String,List<String>> perRecipientHeaderValue = perRecipientHeaders.get(recipient);
-            for (String perRecipientHeader : perRecipientHeaderValue.keySet()) {
-                for (String perRecipientValue : perRecipientHeaderValue.get(perRecipientHeader)) {
-                    mailBuilder.addHeaderForRecipient(Header.builder()
-                            .name(perRecipientHeader)
-                            .value(perRecipientValue)
-                            .build(), recipientKey);
-                }
-            }
-        }
-
-        mailRepository.store(mailBuilder.build());
+            .size(42424242)
+            .addHeaderForRecipient(Header.builder()
+                    .name("headerName1")
+                    .value("value1")
+                    .build(), recipientHeaderAddress)
+            .addHeaderForRecipient(Header.builder()
+                    .name("headerName1")
+                    .value("value2")
+                    .build(), recipientHeaderAddress)
+            .addHeaderForRecipient(Header.builder()
+                    .name("headerName2")
+                    .value("value3")
+                    .build(), recipientHeaderAddress)
+            .addHeaderForRecipient(Header.builder()
+                    .name("headerName2")
+                    .value("value4")
+                    .build(), recipientHeaderAddress)
+            .build();
 
-        when()
-            .get(URL_ESCAPED_MY_REPO + "/mails/" + name
-                    + "?additionalFields=attributes,headers,body,messageSize,perRecipientsHeaders")
-        .then()
-            .statusCode(HttpStatus.OK_200)
-            .body("name", is(name))
-            .body("headers", Matchers.allOf(headers
-                    .keySet()
-                    .stream()
-                    .map((headerName) -> Matchers.hasEntry(headerName, headers.get(headerName)))
-                    .collect(Collectors.toList())))
-            .body("body", is(body))
-            .body("messageSize", is(messageSize))
-            .body("attributes", is(attributes))
-            .body("perRecipientsHeaders", is(perRecipientHeaders));
+        mailRepository.store(mail);
+
+        String jsonAsString =
+            given()
+                .parameters("additionalFields", "attributes,headers,body,messageSize,perRecipientsHeaders")
+            .when()
+                .get(URL_ESCAPED_MY_REPO + "/mails/" + name)
+            .then()
+            .log().all()
```

Comment:
`.log().all()` is a debug statement and should be removed.

Use `logIfValidationFails` at the class level instead.